java/core/src/java/org/apache/orc/Reader.java

```diff
@@ -114,7 +114,7 @@
    * Get the statistics about the columns in the file.
    * @return the information about the column
    */
-  ColumnStatistics[] getStatistics();
+  ColumnStatistics[] getStatistics() throws IOException;
```

Comment:
is this required? when does this throw exception?

Reply:
Good catch. I can't add a throw here without breaking backwards compatibility. In general, we won't throw here, but if the key provider is flaky, we can get an exception. I guess I'll make it an unchecked exception.